h/views/api_auth.py

```diff
@@ -31,6 +32,43 @@ def __init__(self, context, request):
     @view_config(request_method='GET',
                  renderer='h:templates/oauth/authorize.html.jinja2')
     def get(self):
+        return self._authorize()
+
+    @view_config(request_method='GET',
+                 request_param='response_mode=web_message',
+                 renderer='h:templates/oauth/authorize.html.jinja2')
+    def get_web_message(self):
+        response = self._authorize()
+
+        if isinstance(response, HTTPFound):
```

Comment:
I think the thinking here is that the redirect response is part of the OAuth spec and the `web_message` response mode is "layered" on top of this as a non-standard addition (albeit inspired by the spec linked in the PR description) for the convenience of browser-based clients.